Add `single` command to RN changelog generator script and refactor

lunaleaps opened the pull request:
### Description
### What issue are you trying to solve?
Create a new command to format a single commit as well as remove the token dependency for getting commit data

### How does this change address the issue?
* Makes the default behavior as a command 
* Create a `single` command
* Extract out util functions

### Test plan

Compared the following command before and after these changes: 
```
# before
node lib/index.js --repo ~/react-native --changelog ~/react-native/CHANGELOG.md --base 0.67-stable --compare main
# after
node lib/index.js --repo ~/react-native --changelog ~/react-native/CHANGELOG.md --base 0.67-stable --compare main --token [mytoken]
```
and verified tests pass via `yarn test:local`

NickGerleman commented on packages/rn-changelog-generator/README.md:
````diff
@@ -61,22 +61,20 @@ git pull
 popd
 ```
 
-Then create a GitHub ‘personal access token’ that can be used by the script to
-fetch commit metadata from the GitHub API.
-
-1. Visit the [token settings page](https://github.com/settings/tokens).
-1. Generate a new token and **only** give it the `public_repo` scope.
-1. Store this token somewhere secure for future usage.
-
````
GitHub may rate limit requests not authenticated with a token. I've noticed this become a problem when running on some public CI services.

Would it be possible to keep the ability to use a token, but to let it be optional?

lunaleaps replied:
Sounds good, can do that

lunaleaps replied:
Updated to make token optional

NickGerleman commented on packages/rn-changelog-generator/src/generator.ts:
```diff
@@ -7,9 +7,13 @@ import fs from "fs";
+  Commit,
+  getChangeMessage,
+  formatCommitLink,
+} from "./utils";
```
Nit: "Utils" files can become a bit of a dumping ground for unrelated functionality. Are there other factorings that could avoid this?

NickGerleman commented on packages/rn-changelog-generator/src/generator.ts:
```diff
@@ -741,80 +617,16 @@ export async function getAllChangelogDescriptions(
-    .version(false)
-    .help("help").argv;
-
+export default function (argv: any) {
```
Really like this change 👍

NickGerleman commented on packages/rn-changelog-generator/src/single.ts:
```diff
@@ -0,0 +1,7 @@
+import { fetchCommit, getChangeMessage } from "./utils";
+
+export default function (argv: any) {
```
Can we do something typesafe?

NickGerleman commented on packages/rn-changelog-generator/src/single.ts:
```diff
@@ -0,0 +1,7 @@
+import { fetchCommit, getChangeMessage } from "./utils";
+
+export default function (argv: any) {
+  fetchCommit(argv.commit).then((commitData) => {
```
Can we do async/await for new code, instead of explicit promise chaining?

NickGerleman commented on packages/rn-changelog-generator/test/utils.test.ts:
```diff
@@ -0,0 +1,106 @@
+import path from "path";
+import util from "util";
+
+const readFile = util.promisify(fs.readFile);
```
Node 10+ (older than we care about) added support for a native promise-based API, `fs.promises`.

A pattern is to do something like:
`import {promises as fs} from "fs";`

NickGerleman commented on packages/rn-changelog-generator/test/utils.test.ts:
```diff
@@ -0,0 +1,106 @@
+console.warn = () => {};
+console.error = () => {};
+
+function requestWithFixtureResponse(fixture: string) {
+  const requestEmitter = new EventEmitter();
+  const responseEmitter = new EventEmitter();
+  (responseEmitter as any).statusCode = 200;
+  (responseEmitter as any).headers = { link: 'rel="next"' };
+  setImmediate(() => {
+    requestEmitter.emit("response", responseEmitter);
+    readFile(path.join(__dirname, "__fixtures__", fixture), "utf-8").then(
+      (data) => {
+        responseEmitter.emit("data", data);
+        responseEmitter.emit("end");
+      }
+    );
+  });
+  return requestEmitter;
+}
```
💯

tido64 commented:
Can you run `yarn change` at the workspace root and add a change log?
![image](https://user-images.githubusercontent.com/4123478/145729736-09949d34-6379-4570-afb6-7dd073aa19f9.png)

lunaleaps commented:
Have addressed all the above comments and is ready for another look! 

Have verified that running the following is the same as on main: 
```
node lib/index.js --repo ~/react-native --changelog ~/react-native/CHANGELOG.md --base v0.66.4 --compare v0.67.0-rc.6 --token TOKEN 
```

and running `single` command looks like: 
```
node lib/index.js single -c 3bd11599c9d6aca5fa29ad291e9ad8b4a353bef4
https://api.github.com/repos/facebook/react-native/commits/3bd11599c9d6aca5fa29ad291e9ad8b4a353bef4
- Update bump-oss-version.js to guide releaser through release actions ([3bd11599c9](https://github.com/facebook/react-native/commit/3bd11599c9d6aca5fa29ad291e9ad8b4a353bef4) by [@lunaleaps](https://github.com/lunaleaps))
{
  doesNotFollowTemplate: false,
  changeCategory: 'general',
  changeType: 'failed',
  fabric: false,
  internal: true,
  turboModules: false
}
```

tido64 reviewed (approved):
LGTM

@NickGerleman: Just give me a thumbs-up and I'll merge this.

tido64 commented:
FYI: https://github.com/microsoft/rnx-kit/releases/tag/%40rnx-kit/rn-changelog-generator%400.2.0
